### ISSUE TITLE
Error setting executable permissions when starting Datadog agent in Docker buildpack

### ISSUE THREAD

**Issue by eist76** [mendix/cf-mendix-buildpack#403]
hello,

this might belong to the docker-mendix-buildpack repo, but due to the recent changes and enhancements regarding Datadog metrics i think this might be related.

i used docker-mendix-buildpack v3.2.0 which utilizes cf-mendix-buildpack v4.15.1 to build a mendix docker image (with DD_API_KEY) successfully with the metrics agent built in (mx-agent-v0.12.0.jar), but when i try to start the container i run into the errors below - both on local docker as well as ECS fargate:

local docker log:
```
INFO: Mendix Cloud Foundry Buildpack v4.15.1 [4cea375] starting...
...
INFO: Starting the Telegraf metrics agent...
ERROR: Starting application failed: Traceback (most recent call last):
  File "/opt/mendix/buildpack/buildpack/start.py", line 184, in <module>
    datadog.run()
  File "/opt/mendix/buildpack/buildpack/datadog.py", line 458, in run
    _set_permissions("{}/*.sh".format(AGENT_DIR))
  File "/opt/mendix/buildpack/buildpack/datadog.py", line 435, in _set_permissions
    os.chmod(exec_file, st.st_mode | stat.S_IEXEC)
PermissionError: [Errno 1] Operation not permitted: '/opt/mendix/build/.local/datadog/lib/redirect-logs.sh'

Traceback (most recent call last):
  File "/opt/mendix/buildpack/buildpack/start.py", line 184, in <module>
    datadog.run()
  File "/opt/mendix/buildpack/buildpack/datadog.py", line 458, in run
    _set_permissions("{}/*.sh".format(AGENT_DIR))
  File "/opt/mendix/buildpack/buildpack/datadog.py", line 435, in _set_permissions
    os.chmod(exec_file, st.st_mode | stat.S_IEXEC)
PermissionError: [Errno 1] Operation not permitted: '/opt/mendix/build/.local/datadog/lib/redirect-logs.sh'
```

when trying to start the image on ECS Fargate i get a similar error:
```
2021-03-12T10:17:18 INFO: Adding uid to /etc/passwd
2021-03-12T10:17:19 INFO: Mendix Cloud Foundry Buildpack v4.15.1 [4cea375] starting...
...
2021-03-12T10:17:19 INFO: Starting the Telegraf metrics agent...
2021-03-12T10:17:19 ERROR: Starting application failed: Traceback (most recent call last):
2021-03-12T10:17:19   File "/opt/mendix/buildpack/buildpack/start.py", line 184, in <module>
2021-03-12T10:17:19     datadog.run()
2021-03-12T10:17:19   File "/opt/mendix/buildpack/buildpack/datadog.py", line 458, in run
2021-03-12T10:17:19     _set_permissions("{}/*.sh".format(AGENT_DIR))
2021-03-12T10:17:19   File "/opt/mendix/buildpack/buildpack/datadog.py", line 435, in _set_permissions
2021-03-12T10:17:19     os.chmod(exec_file, st.st_mode | stat.S_IEXEC)
2021-03-12T10:17:19 PermissionError: [Errno 1] Operation not permitted: '/opt/mendix/build/.local/datadog/lib/run-datadog.sh'
2021-03-12T10:17:22 Traceback (most recent call last):
2021-03-12T10:17:22   File "/opt/mendix/buildpack/buildpack/start.py", line 184, in <module>
2021-03-12T10:17:22     datadog.run()
2021-03-12T10:17:22   File "/opt/mendix/buildpack/buildpack/datadog.py", line 458, in run
2021-03-12T10:17:22     _set_permissions("{}/*.sh".format(AGENT_DIR))
2021-03-12T10:17:22   File "/opt/mendix/buildpack/buildpack/datadog.py", line 435, in _set_permissions
2021-03-12T10:17:22     os.chmod(exec_file, st.st_mode | stat.S_IEXEC)
2021-03-12T10:17:22 PermissionError: [Errno 1] Operation not permitted: '/opt/mendix/build/.local/datadog/lib/run-datadog.sh'
```

Please let me know if this issue should be submitted somewhere else.
any hints or feedback is appreciated.
Regards,
Eike

**Comment by markbastiaans** [comment]
Hi @eist76 - this is definitely an issue for the Docker buildpack. We target Cloud Foundry with this buildpack and in Cloud Foundry, there are no issues with the permission commands we are required to run.

**Comment by markbastiaans** [comment]
Hi @eist76 - we've created and tested #410 together with the Docker buildpack maintainers. It's been merged into the `develop` branch, so expect the fix to be in the next Cloud Foundry buildpack release, and subsequently to be picked up by the Docker buildpack.

Thanks for your patience!

**Comment by zlogic** [comment]
This issue can be fixed by upgrading to Docker Buildpack v3.3.0 https://github.com/mendix/docker-mendix-buildpack/releases/tag/v3.3.0
